specs.md

```diff
@@ -175,11 +177,23 @@ BMS-38 | AIR states transmission | The BMS must transmit the open/closed states
 ### BMS Init State <a name="BMS_INIT"></a>
 ID | Title | Description | Associated Competition Rule(s)
 --- | --- | --- | ---
-BMS-12 | Precharge | - The BMS must wait for 5 seconds after boot, then wait for the closing of the AIR-, to execute the precharge sequence. <br/> - The BMS must precharge the inverter/charger capacitors to at least 98% of the accumulator voltage for extra safety margin. <br/> - Upon a successful precharge, the BMS must close the AIR+. <br/> <br/> Upon a precharge failure, the BMS must throw an AIR shutdown fault. A precharge failure occurs when: <br/> - The TS (tractive system) bus voltage does not rise within the allotted time. <br/> - The TS bus voltage rises too quickly. ([TODO: calculate constants](https://github.com/UBCFormulaElectric/Consolidated-Firmware/issues/515))| EV.6.9.1
 BMS-35 | SoC retrieval | The BMS must retrieve SoC from three different EEPROM regions, and use a voting algorithm to identify which data is correct, in case of data corruption. In the case of data corruption, set the SoC to 50%. 
 BMS-13 | Default State | The BMS state machine must begin in the init state by default.
 BMS-15 | Exiting the init state and entering the charge state | Upon a successful precharge, the BMS must enter the charge state if the charger is connected.
 BMS-16 | Exiting the init state and entering the drive state | Upon a successful precharge, the BMS must enter the drive state if the charger is disconnected.
+BMS-x  | Exiting the init state and entering the AIR-Open state | Upon waiting for 5 seconds after boot, the BMS must enter the AIR-Open state. 
+
+### BMS AIR-Open State <a name="BMS_AIR_OPEN"></a>
+ID | Title | Description | Associated Competition Rule(s)
+--- | --- | --- | ---
+BMS-x | Entering the AIR-Open state | The BMS must only enter the AIR-Open state from the init state after waiting for 5 seconds after boot. 
+BMS-x | Exiting the AIR-Open state and entering the precharge state | The BMS must only enter the precharge state after AIR- is closed.
+
+### BMS Precharge State <a name="BMS_PRECHARGE"></a>
+ID | Title | Description | Associated Competition Rule(s)
+--- | --- | --- | ---
+BMS-x | Precharge | - The BMS must precharge the inverter/charger capacitors to at least 98% of the accumulator voltage for extra safety margin. <br/> - Upon a successful precharge, the BMS must close the AIR+. <br/> <br/> - Upon a precharge failure, the BMS must throw an AIR shutdown fault. A precharge failure occurs when: <br/> - The TS (tractive system) bus voltage does not rise within the allotted time. <br/> - The TS bus voltage rises too quickly. ([TODO: calculate constants](https://github.com/UBCFormulaElectric/Consolidated-Firmware/issues/515))| EV.6.9.1 The BMS must precharge the inverter/charger capacitors to at least 98% of the accumulator voltage for extra safety margin.
```

Comment:
i would separate this into two definitions:
- successful precharge
- failed precharge
and add a spec to exit the precharge state and enter the fault state (if we want to reset precharge through pressing the latch reset button AND cycling the TSMS) or enter the init state (if we only want to cycle the TSMS)

remember to specify that the minimum time to precharge is the time associated with 90% charged to avoid dangerous closings of the AIRs due to noise

remember to add a heavy extra filter for precharge voltage sensing specifically

Reply:
> @kirons96  or enter the init state (if we only want to cycle the TSMS)


Is there a situation where we will do this? If we don't reset the latch, BMS_ISO, IMD_ISO and BSPD_ISO can still all open the AIR loop which will prevent us from entering the precharge state if we only cycle the TSMS.

We also don't have a way of getting the latch's outputs to figure when we should decide to reset and cycle TSMS, or only cycle TSMS.

Reply:
@kirons96 Alright just confirming after FB convo that we will be resetting the latch before cycling.

Reply:
see my other comments about the latch

to clarify, my comment was outlining two separate options of precharge fault behavior:
1. enter the init state (yolo)
2. enter the fault state (if we want to reset precharge through pressing the latch reset button AND cycling the TSMS)

we went with 1

Reply:
slatt